modules/friendlytag.js

```diff
@@ -183,6 +183,47 @@ Twinkle.tag.callback = function friendlytagCallback() {
 		case 'redirect':
 			Window.setTitle('Redirect tagging');
 
+			// Utility function
+			// https://stackoverflow.com/a/2548133/3480193
+			var endsWith = function(str, suffix) {
+				return str.indexOf(suffix, str.length - suffix.length) !== -1;
+			};
+
+			// If a tag has a restriction for this namespace or title, return true, so that we know not to display it in the list of check boxes.
+			var isRestricted = function(item) {
+				if (typeof item.restriction === 'undefined') {
+					return false;
+				}
+				var namespace = mw.config.get('wgNamespaceNumber');
+				switch (item.restriction) {
+					case 'insideMainspaceOnly':
+						if (namespace !== 0) {
+							return true;
+						}
+						break;
+					case 'outsideUserspaceOnly':
+						if (namespace === 2 || namespace === 3) {
+							return true;
+						}
+						break;
+					case 'insideTalkNamespaceOnly':
+						if (namespace % 2 !== 1 || namespace < 0) {
+							return true;
+						}
+						break;
+					case 'disambiguationPagesOnly':
+						if (!endsWith(mw.config.get('wgPageName'), '_(disambiguation)')) {
```

Comment:
This wouldn't cover all disambig pages (eg. [Android](https://en.wikipedia.org/wiki/Android)). Checking wgCategories for "All disambiguation pages" should be exhaustive, I think. Another option would have been checking for `$('.dmbox').length` – but that will only work while viewing a page (not in edit/history mode).

Reply:
I think it'd be OK to leave as-is. It's counter intuitive, but the [documentation for {{R to disambiguation page}}](https://en.wikipedia.org/wiki/Template:R_to_disambiguation_page) says "this template should only appear on a redirect page that has "(disambiguation)" at the end of its title".